src/tests/networking_tests.rs

```diff
@@ -670,19 +670,248 @@ pub mod net_tests {
     }
 
     #[test]
-    pub fn ut_lind_net_poll() {
-        //acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+    pub fn ut_lind_net_poll_bad_input() {
+        // this test is used for testing poll with some error/edge cases
+
+        // acquiring a lock on TESTMUTEX prevents other tests from running concurrently,
+        // and also performs clean env setup
+        let _thelock = setup::lock_and_init();
+
+        let cage = interface::cagetable_getref(1);
+
+        // error case 1: invalid file descriptor
+        // constrcut a PollStruct with invalid fd (10)
+        let mut polled = vec![interface::PollStruct {
+            fd: 10,
+            events: POLLIN,
+            revents: 0,
+        }];
+
+        // exactly one fd should have non-zero revents field
+        assert_eq!(cage.poll_syscall(&mut polled.as_mut_slice(), None), 1);
+        // and its revents should be set to POLLNVAL
+        assert_eq!(polled[0].revents, POLLNVAL);
+
+        // error case 2: negative file descriptor should be ignored
+        // constrcut a PollStruct with negative fd
+        let mut polled = vec![interface::PollStruct {
+            fd: -1,
+            events: POLLIN,
+            revents: 0,
+        }];
+
+        // the fd should be ignored, so no error is expected
+        assert_eq!(
+            cage.poll_syscall(
+                &mut polled.as_mut_slice(),
+                Some(interface::RustDuration::ZERO)
+            ),
+            0
+        );
+        // revents should be 0
+        assert_eq!(polled[0].revents, 0);
+
+        // edge case: revents should always be cleared
+        // create a file
+        let filefd = cage.open_syscall("/netpolltest.txt", O_CREAT | O_EXCL | O_RDWR, S_IRWXA);
+        assert!(filefd > 0);
+        // create a pipe
```

Comment:
How is the pipe being used here?

Reply:
pipe fd is the most easy to use file descriptor (I think) that has a ready/unready state. Other types of file descriptors are either too complicated to create (e.g. socket fds), or just do not have an unready state (file fd).